chips/nrf52/src/power.rs

```diff
@@ -0,0 +1,351 @@
+            Detected = 1
+        ],
+        DIF OFFSET(18) NUMBITS(1) [
+            Detected = 1
+        ],
+        NFC OFFSET(19) NUMBITS(1) [
+            Detected = 1
+        ],
+        VBUS OFFSET(20) NUMBITS(1) [
+            Detected = 1
+        ]
+    ],
+
+    PowerFailure [
+        POF OFFSET(0) NUMBITS(1) [
+            Disabled = 0,
+            Enabled = 1
+        ],
+        THRESHOLD OFFSET(1) NUMBITS(4) [
+            V17 = 4,
+            V18 = 5,
+            V19 = 6,
+            V20 = 7,
+            V21 = 8,
+            V22 = 9,
+            V23 = 10,
+            V24 = 11,
+            V25 = 12,
+            V26 = 13,
+            V27 = 14,
+            V28 = 15
+        ],
+        THRESHOLDVDDH OFFSET(8) NUMBITS(4) [
+            V27 = 0,
+            V28 = 1,
+            V29 = 2,
+            V30 = 3,
+            V31 = 4,
+            V32 = 5,
+            V33 = 6,
+            V34 = 7,
+            V35 = 8,
+            V36 = 9,
+            V37 = 10,
+            V38 = 11,
+            V39 = 12,
+            V40 = 13,
+            V41 = 14,
+            V42 = 15
+        ]
+    ],
+
+    Byte [
+        VALUE OFFSET(0) NUMBITS(8)
+    ],
+
+    UsbRegStatus [
+        VBUSDETECT OFFSET(0) NUMBITS(1),
+        OUTPUTRDY OFFSET(1) NUMBITS(1)
+    ],
+
+    MainSupply [
+        MAINREGSTATUS OFFSET(0) NUMBITS(1) [
+            Normal = 0,
+            High = 1
+        ]
+    ],
+
+    RamPower [
+        S0POWER OFFSET(0) NUMBITS(1),
+        S1POWER OFFSET(1) NUMBITS(1),
+        S2POWER OFFSET(2) NUMBITS(1),
+        S3POWER OFFSET(3) NUMBITS(1),
+        S4POWER OFFSET(4) NUMBITS(1),
+        S5POWER OFFSET(5) NUMBITS(1),
+        S6POWER OFFSET(6) NUMBITS(1),
+        S7POWER OFFSET(7) NUMBITS(1),
+        S8POWER OFFSET(8) NUMBITS(1),
+        S9POWER OFFSET(9) NUMBITS(1),
+        S10POWER OFFSET(10) NUMBITS(1),
+        S11POWER OFFSET(11) NUMBITS(1),
+        S12POWER OFFSET(12) NUMBITS(1),
+        S13POWER OFFSET(13) NUMBITS(1),
+        S14POWER OFFSET(14) NUMBITS(1),
+        S15POWER OFFSET(15) NUMBITS(1),
+        S0RETENTION OFFSET(16) NUMBITS(1),
+        S1RETENTION OFFSET(17) NUMBITS(1),
+        S2RETENTION OFFSET(18) NUMBITS(1),
+        S3RETENTION OFFSET(19) NUMBITS(1),
+        S4RETENTION OFFSET(20) NUMBITS(1),
+        S5RETENTION OFFSET(21) NUMBITS(1),
+        S6RETENTION OFFSET(22) NUMBITS(1),
+        S7RETENTION OFFSET(23) NUMBITS(1),
+        S8RETENTION OFFSET(24) NUMBITS(1),
+        S9RETENTION OFFSET(25) NUMBITS(1),
+        S10RETENTION OFFSET(26) NUMBITS(1),
+        S11RETENTION OFFSET(27) NUMBITS(1),
+        S12RETENTION OFFSET(28) NUMBITS(1),
+        S13RETENTION OFFSET(29) NUMBITS(1),
+        S14RETENTION OFFSET(30) NUMBITS(1),
+        S15RETENTION OFFSET(31) NUMBITS(1)
+    ]
+];
+
+pub struct Power<'a> {
+    registers: StaticRef<PowerRegisters>,
+    usb_client: OptionalCell<&'a dyn PowerClient>,
+}
+
+pub enum MainVoltage {
+    Normal = 0,
+    High = 1,
```

Comment:
What does `High` mean?

Reply:
That's the terminology used by Nordic in the data sheet. Added some explanation.